app/controllers/auth0_controller.rb

```diff
@@ -1,16 +1,30 @@
 class Auth0Controller < ApplicationController
   def callback
-    # This stores all the user information that came from Auth0
-    # and the IdP
-
+    
     session[:userinfo] = request.env['omniauth.auth']
+    @user_info = session[:userinfo]
+    
+    params[:token] = @user_info.credentials["token"]
+    params[:email] = @user_info.info["email"]
 
-    # Redirect to the URL you want after successfull auth
-    redirect_to '/blogs'
+    users = User.all
+    if users.find_by_email(params[:email]).nil?
+      new_user
+    else
+      redirect_to '/blogs'
+    end
+    
   end
 
-  def user
-    @user = session[:userinfo]
+  def new_user
+      @user = User.new(pass_info) 
```

Comment:
can we please use ``` user_params ```
this is a little more descriptive as to what you are passing